openff/toolkit/_tests/test_topology.py

```diff
@@ -754,7 +754,9 @@ def test_to_from_openmm_hierarchy_metadata(self):
                 roundtrip = roundtrip_atom.metadata["residue_number"]
                 assert original == roundtrip
             else:
-                assert roundtrip_atom.metadata["residue_number"] == 0
+                # OpenFF lets residue numbers be str or int, but this is str in OpenMM's
+                # representation, so faithful round-trip is not practical
```

Comment:
(blocking) Good explanation but confusing when read outside the context of this PR, let's make sure to delete this comment before merging